modules/ecs-service/main.tf

```diff
@@ -0,0 +1,696 @@
+      }]
+
+      # Environment Variables and Secrets are both string maps with
+      # the same key/value structure. They are mapped to the appropriate
+      # structure for the container definition
+      environment = [for k, v in var.ecs_container_environment_variables : { name = k, value = v }]
+      secrets     = [for k, v in var.ecs_container_secrets : { name = k, valueFrom = v }]
+
+      logConfiguration = var.enable_container_logs ? local.log_configuration : null
+
+      # healthCheck = !var.create_scheduled_task ? local.healthCheck : null
+    }
+  ])
+
+  requires_compatibilities = var.create_scheduled_task ? ["FARGATE"] : ["EC2"]
+
+  runtime_platform {
+    operating_system_family = "LINUX"
+    cpu_architecture        = "X86_64"
+  }
+
+  depends_on = [
+    aws_cloudwatch_log_group.task
+  ]
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+
+# -------------------------------------------
+# CREATE THE CLOUDWATCH LOG GROUP FOR THE TASK
+# -------------------------------------------
+
+# tfsec:ignore:aws-cloudwatch-log-group-customer-key
+resource "aws_cloudwatch_log_group" "task" {
+  count = var.enable_container_logs ? 1 : 0
+
+  name              = local.log_group_name
+  retention_in_days = 30
+
+  lifecycle {
+    prevent_destroy = false
+  }
+}
+
+
+# -------------------------------------------
+# CREATE THE TASK IAM ROLE
+# -------------------------------------------
+
+resource "aws_iam_role" "task" {
+  count = length(var.ecs_task_role_policy_arns) > 0 ? 1 : 0
+
+  name_prefix = "${var.ecs_service_name}-task"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [{
+      Effect = "Allow"
+      Principal = {
+        Service = "ecs-tasks.amazonaws.com"
+      }
+      Action = "sts:AssumeRole"
+      Condition = {
+        test     = "ArnLike"
+        variable = "aws:SourceArn"
+        values = [
+          "arn:aws:ecs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:*"
+        ]
+      }
+      Condition = {
+        test     = "StringEquals"
+        variable = "aws:SourceAccount"
+        values = [
+          data.aws_caller_identity.current.account_id
+        ]
+      }
+    }]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "task" {
+  count = length(var.ecs_task_role_policy_arns) > 0 ? length(var.ecs_task_role_policy_arns) : 0
+
+  role       = aws_iam_role.task[0].name
+  policy_arn = var.ecs_task_role_policy_arns[count.index]
+}
+
+
+# -------------------------------------------
+# CREATE THE TASK EXECUTION IAM ROLE
+# -------------------------------------------
+
+resource "aws_iam_role" "task_execution" {
+  name_prefix = "${var.ecs_service_name}-execution"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [{
+      Effect = "Allow"
+      Principal = {
+        Service = "ecs-tasks.amazonaws.com"
+      }
+      Action = "sts:AssumeRole"
+    }]
+  })
+}
```

Comment:
You can use a generic aws_iam_policy_document called `assume_role` that can be applied to multiple roles. Within aws_iam_role you can specify additional configureation as needed. In your case here it's a generic assume role. I do this in #31

Reply:
I don't see an example of extending the assume role policy in their documentation. The difference between this one, and the task role is that this one doesn't have conditions like the task roles needs.